### ISSUE TITLE
Repository#workdir and Repository#path should not include trailing slash

### ISSUE THREAD

**Issue by mojavelinux** [nodegit/nodegit#1461]
It's very unconventional to add a trailing slash to the directory name returned by these methods. It also differs from the values reported by the native git client.

Case in point.

*Native git*
```
$ git rev-parse --absolute-git-dir
/home/user/projects/nodegit/.git
```

*nodegit*
```
repo.path()
//=> /home/user/projects/nodegit/.git/
```

*Native git*
```
$ git worktree list
# /home/user/projects/nodegit    210d02a [master]
```

*nodegit*
```
repo.workdir()
//=> /home/user/projects/nodegit/
```

I believe these values should either be changed to match the native git client or this difference should be highlighted in the documentation.

**Comment by rcjsuen** [comment]
Is this a problem with NodeGit or libgit2?

**Comment by mojavelinux** [comment]
I don't know. I only use libgit2 via nodegit. I don't see nodegit appending a slash, so I suppose it could be coming from libgit2.

**Comment by mojavelinux** [comment]
I do notice at least one place where nodegit is assuming the trailing slash is present, so those would need to be updated if there were to be a change.

**Comment by rcjsuen** [comment]
@mojavelinux Yes, please open an issue with libgit2 and see what they say. We'll leave this open for now.

**Comment by mojavelinux** [comment]
Done. See https://github.com/libgit2/libgit2/issues/4583

How can I tell which version of libgit2 is used by nodegit?

**Comment by rcjsuen** [comment]
@mojavelinux That's not super straightforward as NodeGit is built off of a [fork](https://github.com/nodegit/libgit2/tree/e002b651a116fe8be07aa56343de32e40bba2679). You're at least on v0.26 anyway.

**Comment by mojavelinux** [comment]
Since libgit2 requires that you specify the version when reporting an issue, I think we should find some way to have a reportable version number so the project takes our requests seriously.

**Comment by rcjsuen** [comment]
@mojavelinux I'm basing my claim off of the [CHANGELOG.md file](https://github.com/nodegit/libgit2/blob/e002b651a116fe8be07aa56343de32e40bba2679/CHANGELOG.md) that is in the fork as well as the [README.md file](https://github.com/nodegit/nodegit/blob/210d02a47e080f4872cc17665ac08feb2e204680/README.md) that is in the root of this repository.
